File: LeetCode/Explore/TopInterview/Amazon/SortingSearching/347_top_k_frequent_elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        mp = dict()
        
        for num in nums:
          mp[num] = mp.get(num, 0) + 1
        
        sp = [k for k, v in sorted(mp.items(), key = lambda x:x[1], reverse=True)]
        
        return sp[:k]

File: LeetCode/Explore/TopInterview/Amazon/SortingSearching/347_top_k_frequent_elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
      unordered_map<int, int> mp;
      priority_queue<pair<int, int> > pq;
      vector<int> ans;
      for (auto &num : nums) {
        mp[num]++;
      }
      for (auto &m : mp) {
        pq.push(make_pair(m.second, m.first));
      }
      while (k) {
        ans.push_back(pq.top().second);
        pq.pop();
        k--;
      }
      return ans;
    }
};
